src/Tribe/iCal.php

```diff
@@ -218,11 +201,29 @@ public static function generate_ical_feed( $post = null ) {
 		if ( $post ) {
 			$events_posts = is_array( $post ) ? $post : array( $post );
 		} else {
-			if ( tribe_is_month() ) {
-				$events_posts = self::get_month_view_events();
+			/**
+			 * Filters the number of upcoming events the iCal feed should export.
+			 *
+			 * This filter allows developer to override the pagination setting and the default value
+			 * to export a number of events that's inferior or superior to the one shown on the page.
+			 * The minimum value is 1.
+			 *
+			 * @param int $count The number of upcoming events that should be exported in the
+			 *                   feed, defaults to 30.
+			 */
+			$count = apply_filters( 'tribe_ical_feed_posts_per_page', $this->feed_default_export_count );
+
+			$count = is_numeric( $count ) && is_int( $count ) && $count > 0 ? $count : $this->feed_default_export_count;
```

Comment:
Lets avoid using the prop `$this-> feed_default_export_count`, and use the method instead?